### PR TITLE
Merge Cosmos with Tendermint

### PR DESCRIPTION
**Pull request by patiee**

### CODE REVIEW

**Review by kruspy — APPROVED** [review]

File: codec.proto [hunk 1]
```diff
@@ -40,11 +40,15 @@ message EventData {
   EventBlock block = 2;
 }
 
+message TransactionData {
+  TxResult   tx    = 1;
+  EventBlock block = 2;
+}
+
```
Comment (lavabyrd) [inline codec.proto:47]: I'm fairly sure you set this file up to be alphabetized before. This should be further down the file if so

File: codec.proto [hunk 2]
```diff
@@ -228,6 +232,7 @@ message TxResult {
   uint32            index  = 2;
   bytes             tx     = 3;
   ResponseDeliverTx result = 4;
+  bytes             hash   = 5;
```
Comment (lavabyrd) [inline codec.proto:235]: does this need to be here?

File: go.mod [hunk 3]
```diff
@@ -1 +1,8 @@
 module github.com/figment-networks/tendermint-protobuf-def
+
+go 1.17
+
+require (
+	github.com/gogo/protobuf v1.3.2
+	google.golang.org/protobuf v1.28.0
+)
```
Comment (lavabyrd) [inline go.mod:8]: do we need these?

File: README.md [hunk 4]
```diff
@@ -13,5 +13,5 @@ make generate
 ## Protobuf formatting
 
 ```
-clang-format '--style={BasedOnStyle: google, ColumnLimit: 130, IndentWidth: 2, SpacesBeforeTrailingComments: 1, AlignConsecutiveAssignments: true, AlignConsecutiveDeclarations: true}' < codec.proto
+clang-format '--style={BasedOnStyle: google, ColumnLimit: 130, IndentWidth: 2, SpacesBeforeTrailingComments: 1, PenaltyBreakComment: 10,AlignConsecutiveAssignments: true, AlignConsecutiveDeclarations: true}' < codec.proto
```
[the remark this answers is not in the document]
Comment (patiee) [reply README.md:16]: we can fix formatting in separeted PR, it's not part of my ticket

File: go.mod (hunk 3 above)
Comment (patiee) [reply go.mod:8]: no, I removed it it #22

**Pull request merged**